onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -93,11 +93,27 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
 #endif
   } else if (provider_name == onnxruntime::kCudaExecutionProvider) {
 #ifdef USE_CUDA
-    OrtCUDAProviderOptions cuda_options;
-    cuda_options.cudnn_conv_algo_search = static_cast<OrtCudnnConvAlgoSearch>(performance_test_config.run_config.cudnn_conv_algo);
-    cuda_options.do_copy_in_default_stream = !performance_test_config.run_config.do_cuda_copy_in_separate_stream;
+      const auto &api = Ort::GetApi();
+      std::string workspace = std::to_string(4ul * 1024ul * 1024ul * 1024ul);
+      std::vector<const char *> option_name = {
+          "do_copy_in_default_stream",
+          "cudnn_conv_algo_search",
+          "prefer_nhwc"
+      };
+      std::vector<const char *> option_value = {
+          !performance_test_config.run_config.do_cuda_copy_in_separate_stream ? "1" : "0",
+          performance_test_config.run_config.cudnn_conv_algo ? "EXHAUSTIVE" : "HEURISTIC",
```

Comment:
Aren't there 3 possible values for `cudnn_conv_algo `- https://github.com/microsoft/onnxruntime/blob/dd3b2cefd6abdd6a0fe6f6701d26d1299f87f1fa/onnxruntime/test/perftest/command_args_parser.cc#L56 ?

Reply:
Yeah sorry this was a rather wacky approach. I refactored this to be much more general